### PR TITLE
errors: remove stack data when sending to sentry

### PR DESCRIPTION
**Pull request by sergiusens**
Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

**Comment by kyrofa** [comment]
(check units)

File: snapcraft/cli/_errors.py [hunk 1]
```diff
@@ -106,7 +106,15 @@ def _submit_trace(exception):
...
-        transport=RequestsHTTPTransport)
+        transport=RequestsHTTPTransport,
+        # Should Raven automatically log frame stacks (including locals)
+        # for all calls as it would for exceptions.
+        auto_log_stacks=True,
+        # Removes all stacktrace context variables. This will cripple the
+        # functionality of Sentry, as you’ll only get raw tracebacks,
+        # but it will ensure no local scoped information is available to the
+        # server.
+        processors=('raven.processors.RemoveStackLocalsProcessor',))
```
Comment (kyrofa) [inline snapcraft/cli/_errors.py]: I'm not 100% sure what this means given the [docs](https://docs.sentry.io/clients/python/advanced/#python-client-config), can you give me an example?

Comment (sergiusens) [reply snapcraft/cli/_errors.py]: While the `SanitizePasswordsProcessor` mentions that it removes all keys which resemble `password`, `secret`, or `api_key` within stacktrace contexts, HTTP bits (such as cookies, POST data, the querystring, and environment), and extra data. I don't want to accidentally fall in a situation where the regex missed out on macaroons, credit cards (not something we do, today at least). To be fair, knowing the value of a variable in our cases most of the time would not make it hard to figure out why the stacktrace happens in most situations.

Comment (kyrofa) [reply snapcraft/cli/_errors.py]: Ah, okay same page now, thank you!

**Pull request merged**